src/pages/index.js

```diff
@@ -1,62 +1,169 @@
 import Card from "../components/Card.js";
-// import { closePopup, openPopup } from "../utils/utils.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupDeleteCard from "../components/PopupDeleteCard.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, config } from "../utils/constants.js";
+import { config } from "../utils/constants.js";
+import Api from "../components/Api.js";
 import "./index.css";
+import { data } from "autoprefixer";
+
+export const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/group-12",
+  headers: {
+    authorization: "cd56466a-c69a-4082-86d8-efb553341f31",
+    "Content-Type": "application/json",
+  },
+});
 
 // profile buttons
 const profileAddBtn = document.querySelector("#profile-add-button");
 const profileEditBtn = document.querySelector("#profile-edit-btn");
+const profileImgEditBtn = document.querySelector("#edit-avatar-btn");
 // forms
 const editForm = document.querySelector("#profile-edit-modal");
 const addForm = document.querySelector("#profile-add-modal");
+const profileImageForm = document.querySelector("#edit-profile-img-modal");
 // form inputs
 const profileNameInput = editForm.querySelector("#profile-name-input");
 const profileDescriptionInput = editForm.querySelector(
   "#profile-description-input"
 );
 
-const userInfo = new UserInfo("#profile-name", "#profile-des");
+const userInfo = new UserInfo(
+  "#profile-name",
+  "#profile-des",
+  "#profile-image"
+);
+let userID;
+api
+  .getUserinfo()
```

Comment:
You need to combine the requests of the user's info and the cards together in `Promise.all` because it might happen so that the `_id` has not been received yet but the cards have and your cards will be rendered not correctly. 
More info is here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

```js

Promise.all([getUserInfo(), getCards()])
// destructure the response
  .then(([userData, cards]) => {
      // set all the data
  })
  .catch(err => {
    // catch possible errors
  });

```